### ISSUE TITLE
Module request: MACS2

### ISSUE THREAD

**Issue by IanCodes** [ewels/MultiQC#476]
**MACS2**

MACS captures the influence of genome complexity to evaluate the significance of enriched ChIP regions, and MACS improves the spatial resolution of binding sites through combining the information of both sequencing tag position and orientation. 

https://github.com/taoliu/MACS

Example output: [macs2_example_output.txt](https://github.com/ewels/MultiQC/files/1071069/macs2_example_output.txt)

Standard log file naming patterns: `*_peaks.xls`

Most important values for the General Stats table:
```
# Redundant rate in treatment: 0.31
# Redundant rate in control: 0.14
```

What should be plotted: Nothing 

Thank you.

**Comment by ewels** [comment]
Thanks @IanCodes! I tidied up your issue a little, hope that's ok (I need to tidy the template really).

Your example log file is called `macs2_example_output.txt` but you say that the standard log filename pattern is `*_peaks.xls`. I guess the latter is for peaks, not log output? Is the file you attached from standard out? Or is it the header from `*_peaks.xls` or something?

Phil

**Comment by IanCodes** [comment]
Sorry for the confusion - I copied a portion of the *peaks.xls output to a new file without keeping to the naming convention.  The peaks are reported in this file.  Log files are not saved to a file by default as it is "error" output.  I capture it using '2>'.  On Sun Grid Engine it is saved in '.e' files.  The *peak.xls file does also contain the useful summary info in the header.

**Comment by ewels** [comment]
Ok great, thanks. So the output is always `samplename_peaks.xls` and it always contains the information that we want to capture (in the file above) at the top of that file. Fab.

Standard error stuff not needed if we have the information we need in a file anyway 👍 

Phil

**Comment by ewels** [comment]
Hi @IanCodes,

I've just had a crack at writing this module. If you could have a go and try to break it, that'd be great!

Cheers,

Phil

**Comment by IanCodes** [comment]
The MACS2 information appears in the General Stats table.  The is no MACS2 table, alongside for example Bowtie2 or Trimmomatic.  Is that intentional?

**Comment by ewels** [comment]
That was intentional, yes. As there's only two values I thought it better to go in the General Stats table alongside values from other modules. I can make a section for it instead if you think it's needed?

Phil

**Comment by ewels** [comment]
ps. More values are parsed and saved to the `multiqc_data` directory:

```
cat multiqc_data/multiqc_macs.yaml
```
```yaml
macs2_output:
  control_fragments_after_filtering: 24925719.0
  control_fragments_total: 29107626.0
  control_max_duplicates: 1.0
  control_redundant_rate: 0.14
  d: 150.0
  fragment_size: 150.0
  treatment_fragments_after_filtering: 17466636.0
  treatment_fragments_total: 25186079.0
  treatment_max_duplicates: 1.0
  treatment_redundant_rate: 0.31
```
(after a run with `-k yaml`).

**Comment by IanCodes** [comment]
I think it would be useful to have the full numbers of fragments before/after filtering, plus the fragment length.  Sorry - more work!

**Comment by ewels** [comment]
Ok, a few ways I could do this.. I could do a bar chart of fragment counts: filtered + not-filtered, stacked. It would be two bars per sample, but that would be fine. I could prepend the sample name with `control_` or `treatment_` (or suffix to group by sample instead of type).

Alternatively I could just do a MACS2 section table with the numbers. Then I could move the redundancy numbers out of General Stats too.. Maybe better?

Fragment size is specified on the command line right? So it's always the same? Or is it dynamically calculated for paired-end data? If it's always one (or a small number) of numbers, I can just write that in the text for the section. If it varies then I need to put it in a table / plot it.

Phil

**Comment by IanCodes** [comment]
I think given that fragment size would have to be provided as a number in a table, you should probably go with your alternative idea of having everything in one table.  The fragment number is determined from the treatment reads, and is usually in the range of 100 to 300.

Thanks,
Ian

**Comment by ewels** [comment]
Ok cool - I tried this but I try to avoid just having loads of tables wherever I can. So I made a plot instead 😆  Sorry!

The report now has fragment length, treatment +  control redundancies in General Statistics, with a bar plot showing the number of fragments (filtered and not filtered).

Hope this is ok - have a play and let me know what you think. I can change to a table (or whatever) if you prefer...

Phil
